components/shared/Search.tsx

```diff
@@ -0,0 +1,33 @@
+'use client';
+
+import searchIcon from '@/public/searchIcon.svg'
+import shortcutIcon from '@/public/shortcutIcon.svg'
+import Image from 'next/image';
+
+export default function Search() {
+  function handleSearch(term: string) {
+    console.log(term);
+  }
+
+  return (
+    <div className="relative flex flex-1 items-center">
+      <Image
+        src={searchIcon}
+        alt="Search Icon"
+        className='ml-3 absolute pointer-events-none'
+      />
+      <input
+        className="w-full rounded-md py-[9px] pl-10 text-sm placeholder:text-paragraph-4-medium placeholder:text-white-500 bg-black-700 focus:outline-none"
+        placeholder="Search..."
+        onChange={(e) => {
+          handleSearch(e.target.value);
+        }}
+      />
+      <Image
+        src={shortcutIcon}
+        alt="Shortcut Icon"
+        className='pointer-events-none flex -ml-8'
```

Comment:
Felt a bit funny setting a negative margin on this icon - didn't know a better workaround though

Reply:
For something like this- I wouldn't use padding left, padding y, and margins with absolute to create the search bar.

Instead, use gaps and flex to keep everything lined up.

```js
 return (
    <div className="flex items-center gap-2 h-8 rounded-md bg-black-700 px-2">
      <Image
        src={searchIcon}
        alt="Search Icon"
        className="pointer-events-none"
      />
      <input
        className="text-sm w-full rounded-md bg-transparent placeholder:text-white-500 focus:outline-none"
        placeholder="Search..."
        onChange={(e) => {
          handleSearch(e.target.value);
        }}
      />
      <Image
        src={shortcutIcon}
        alt="Shortcut Icon"
        className="pointer-events-none"
      />
    </div>
```    
I'm removed the `placeholder:text-paragraph-4-medium` did you mean `placeholder:paragraph-4-medium`?

But the idea here was just to make the div the color we want, and wrap it around the icons and searchbar with a flex. This gives us one place to control the gap, padding, and size.

![image](https://github.com/DDVVPP/GitNote/assets/4108484/1a005f91-38fb-49b7-a02d-92d719ad7503)